### ISSUE TITLE
`cargo test` fails with in-work unicode diagnostics for Cargo/rust

### ISSUE THREAD

**Issue by epage** [obi1kenobi/cargo-semver-checks#1577]
### Steps to reproduce the bug with the above code

Run
```console
$ cargo +nightly test -Zrustc-unicode
```

### Actual Behaviour

```
---- snapshot_tests::workspace_baseline_conditional_compile_error stdout ----
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot file: src/../test_outputs/snapshot_tests/cargo_semver_checks__snapshot_tests__workspace_baseline_conditional_
compile_error-output.snap
Snapshot: workspace_baseline_conditional_compile_error-output
Source: src/snapshot_tests.rs:217
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Expression: result
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────
   21    21 │ error: running cargo-doc on crate 'error' failed with output:
   22    22 │ -----
   23    23 │  Documenting error v0.1.0 ([ROOT]/test_crates/manifest_tests/workspace_baseline_conditional_compile_erro
r/old/error)
   24    24 │ error: This crate has a compiler error.
   25       │- --> [ROOT]/test_crates/manifest_tests/workspace_baseline_conditional_compile_error/old/error/src/lib.rs
:7:1
   26       │-  |
   27       │-7 | compile_error!("This crate has a compiler error.");
   28       │-  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
         25 │+  ╭▸ [ROOT]/test_crates/manifest_tests/workspace_baseline_conditional_compile_error/old/error/src/lib.rs
:7:1
         26 │+  │
         27 │+7 │ compile_error!("This crate has a compiler error.");
         28 │+  ╰╴━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   29    29 │
   30    30 │ error: could not document `error`
   31    31 │
   32    32 │ -----
────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────
To update snapshots run `cargo insta review`
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'snapshot_tests::workspace_baseline_conditional_compile_error' (1618070) panicked at /home/epage/.cargo/registr
y/src/index.crates.io-1949cf8c6b5b557f/insta-1.46.3/src/runtime.rs:719:13:
snapshot assertion for 'workspace_baseline_conditional_compile_error-output' failed in line 217
```

### Expected Behaviour

Tests pass

### Generated System Information

```
System information:
-------------------
#### Software version

cargo-semver-checks 0.46.0 (7197800)

#### Operating system

- OS: Linux (Pop!_OS 24.04)
- Kernel: 6.17.9-76061709-generic

#### Command-line

```bash
target/debug/cargo-semver-checks semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.93.1 (083ac5135 2025-12-15)
```

#### Compile time information

- Profile: debug
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

Cargo build configuration:
--------------------------
[registry]
global-credential-providers = [["cargo:token"], ["cargo:macos-keychain"], ["cargo:wincred"]]

Please file an issue on GitHub reporting your bug.
Consider adding the diagnostic information above, either manually or automatically through the link below:

https://github.com/obi1kenobi/cargo-semver-checks/issues/new?labels=C-bug&template=3-bug-report.yml&sys-info=%23%23%23
%23%20Software%20version%0A%0Acargo-semver-checks%200.46.0%20%287197800%29%0A%0A%23%23%23%23%20Operating%20system%0A%0
A-%20OS%3A%20Linux%20%28Pop%21_OS%2024.04%29%0A-%20Kernel%3A%206.17.9-76061709-generic%0A%0A%23%23%23%23%20Command-lin
e%0A%0A%60%60%60bash%0Atarget%2Fdebug%2Fcargo-semver-checks%20semver-checks%20--bugreport%20%0A%60%60%60%0A%0A%23%23%2
3%23%20cargo%20version%0A%0A%60%60%60%0A%3E%20cargo%20-V%20%0Acargo%201.93.1%20%28083ac5135%202025-12-15%29%0A%60%60%6
0%0A%0A%23%23%23%23%20Compile%20time%20information%0A%0A-%20Profile%3A%20debug%0A-%20Target%20triple%3A%20x86_64-unkno
wn-linux-gnu%0A-%20Family%3A%20unix%0A-%20OS%3A%20linux%0A-%20Architecture%3A%20x86_64%0A-%20Pointer%20width%3A%2064%0
A-%20Endian%3A%20little%0A-%20CPU%20features%3A%20fxsr%2Csse%2Csse2%0A-%20Host%3A%20x86_64-unknown-linux-gnu%0A%0A&bui
ld-config=%5Bregistry%5D%0Aglobal-credential-providers%20%3D%20%5B%5B%22cargo%3Atoken%22%5D%2C%20%5B%22cargo%3Amacos-k
eychain%22%5D%2C%20%5B%22cargo%3Awincred%22%5D%5D%0A
```

### Build Configuration

_No response_

### Additional Context

Tests pass when running
```console
$ CARGO_TERM_UNICODE=false cargo +nightly test -Zrustc-unicode
```

Likely, the integration tests should set this environment variable when calling the `cargo` process

**Comment by obi1kenobi** [comment]
@codex please set `CARGO_TERM_UNICODE=false` in integration tests
